plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1155,50 +1159,90 @@ public void renameTable(ConnectorSession session, ConnectorTableHandle tableHand
         catalog.renameTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName(), newTable);
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public void setTableProperties(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Optional<Object>> properties)
     {
         IcebergTableHandle table = (IcebergTableHandle) tableHandle;
         BaseTable icebergTable = (BaseTable) catalog.loadTable(session, table.getSchemaTableName());
 
         transaction = icebergTable.newTransaction();
-        UpdateProperties updateProperties = transaction.updateProperties();
+        Optional<UpdateProperties> updateProperties = Optional.empty();
 
         for (Map.Entry<String, Optional<Object>> propertyEntry : properties.entrySet()) {
             String trinoPropertyName = propertyEntry.getKey();
             Optional<Object> propertyValue = propertyEntry.getValue();
 
             switch (trinoPropertyName) {
                 case FILE_FORMAT_PROPERTY:
-                    updateProperties.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg());
+                    updateProperties = updateProperties.or(() -> Optional.of(transaction.updateProperties()))
+                                    .map(update -> update.defaultFormat(((IcebergFileFormat) propertyValue.orElseThrow()).toIceberg()));
                     break;
                 case FORMAT_VERSION_PROPERTY:
                     // UpdateProperties#commit will trigger any necessary metadata updates required for the new spec version
-                    updateProperty(updateProperties, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion));
+                    updateProperties = updateProperties.or(() -> Optional.of(transaction.updateProperties()))
+                            .map(update -> updateProperty(update, FORMAT_VERSION, propertyValue, formatVersion -> Integer.toString((int) formatVersion)));
+                    break;
+                case PARTITIONING_PROPERTY:
+                    // Partition updating is handled after the UpdateProperties is committed
                     break;
                 default:
-                    // TODO: Support updating partitioning https://github.com/trinodb/trino/issues/12174
                     throw new TrinoException(NOT_SUPPORTED, "Updating the " + trinoPropertyName + " property is not supported");
             }
         }
 
         try {
-            updateProperties.commit();
+            updateProperties.ifPresent(PendingUpdate::commit);
```

Comment:
updateProperties is Optional only so that we skip commit when nothing to update. it harms readability otherwise.

and it seems redundant.

Reply:
Yeah, agreed. Reworked this method a bit to ovoid the optional